lightning/src/util/events.rs

```diff
@@ -254,6 +255,14 @@ pub enum Event {
 		channel_id: [u8; 32],
 		/// The reason the channel was closed.
 		reason: ClosureReason
+	},
+	/// Used to indicate to the user that they can abandon the funding transaction and recycle the
+	/// inputs for another purpose.
+	DiscardFunding {
+		/// The channel_id of the channel which has been closed.
+		channel_id: [u8; 32],
+		/// The full transaction received from the user
+		transaction: Option<OutPoint>
```

Comment:
It would be nice to use the full transaction instead - ie get a copy of `Channel::funding_transaction` and put it in the event instead of only the outpoint.